diff --git src/main/java/eu/openanalytics/containerproxy/auth/impl/saml/SAMLConfiguration.java src/main/java/eu/openanalytics/containerproxy/auth/impl/saml/SAMLConfiguration.java
index ccda148..2396b43 100644
--- src/main/java/eu/openanalytics/containerproxy/auth/impl/saml/SAMLConfiguration.java
+++ src/main/java/eu/openanalytics/containerproxy/auth/impl/saml/SAMLConfiguration.java
@@ -56,7 +56,7 @@ import org.springframework.security.saml.SAMLBootstrap;
 import org.springframework.security.saml.SAMLCredential;
 import org.springframework.security.saml.SAMLEntryPoint;
 import org.springframework.security.saml.SAMLProcessingFilter;
-import org.springframework.security.saml.context.SAMLContextProviderImpl;
+import org.springframework.security.saml.context.SAMLContextProviderLB;
 import org.springframework.security.saml.key.EmptyKeyManager;
 import org.springframework.security.saml.key.JKSKeyManager;
 import org.springframework.security.saml.key.KeyManager;
@@ -232,8 +232,15 @@ public class SAMLConfiguration {
 	}
 
 	@Bean
-	public SAMLContextProviderImpl contextProvider() {
-		return new SAMLContextProviderImpl();
+	public SAMLContextProviderLB contextProvider() {
+		SAMLContextProviderLB provider = new SAMLContextProviderLB();
+		provider.setContextPath("/");
+		provider.setIncludeServerPortInRequestURL(false);
+		provider.setScheme("https");
+		String serverName = environment.getProperty("proxy.saml.server-name");
+		provider.setServerName(serverName);
+		provider.setServerPort(443);
+		return provider;
 	}
 
 	@Bean
